tests/acceptance/25_cf-execd/mail_include_and_exclude_filters.cf

```diff
@@ -0,0 +1,46 @@
+# Tests that combining include and exclude mail filters works, IOW only lines
+# matching the include filter should be printed, unless they also match the
+# exclude filter.
+
+body common control
+{
+    inputs => { "../default.cf.sub", "mailfilter_common.cf.sub" };
+    bundlesequence => { default($(this.promise_filename)) };
+}
+
+bundle agent init
+{
+  vars:
+      "reports"  slist => {
+                            "CorrectString1",
+                            "CorrectString2",
+                            "WrongString1",
+                            "WrongString2",
+                            "WrongString3",
+                          };
+      "includes" slist => {
+                            "R: CorrectString1", ".*CorrectString[2]",
+                            "R: WrongString1", ".*WrongString[2]"
+                          };
+      "excludes" slist => {
+                            "R: WrongString1", ".*WrongString[2]",
+                            # Tests anchoring.
+                            "orrectString1"
```

Comment:
Yes, we are testing that a partial match does not match. I can improve the comment a bit to clarify this.